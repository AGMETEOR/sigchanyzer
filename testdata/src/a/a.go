package p

import (
	"os"
	ao "os"
	"os/signal"
)

func f() {
	c := make(chan os.Signal, 1) // ok
	signal.Notify(c, os.Interrupt)
	_ = <-c
}

func g() {
	c := make(chan os.Signal) // want "unbuffer os.Signal channel"
	signal.Notify(c, os.Interrupt)
	_ = <-c
}

func h() {
	c := make(chan ao.Signal) // want "unbuffer os.Signal channel"
	signal.Notify(c, os.Interrupt)
	_ = <-c
}
