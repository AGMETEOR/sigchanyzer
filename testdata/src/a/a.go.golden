func g() {
	c := make(chan os.Signal, 1)
	signal.Notify(c, os.Interrupt)
	_ = <-c
}

func h() {
	c := make(chan ao.Signal, 1)
	signal.Notify(c, os.Interrupt)
	_ = <-c
}
